#! /bin/sh -e
## 30_pipermail_threads.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: 
## DP: non-appropriate output.  Change that.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@

--- mailman-2.1.4.orig/Mailman/Archiver/pipermail.py
+++ mailman-2.1.4/Mailman/Archiver/pipermail.py
@@ -115,9 +115,9 @@
         parentID = article.parentID
         if parentID is not None and self.articleIndex.has_key(parentID):
             parent = self.getArticle(archive, parentID)
-            myThreadKey = parent.threadKey + article.date + '-'
+            myThreadKey = parent.threadKey + article.date + '/' + article.msgid + '-'
         else:
-            myThreadKey = article.date + '-'
+            myThreadKey = article.date + '/' + article.msgid + '-'
         article.threadKey = myThreadKey
         key = myThreadKey, article.msgid
         self.setThreadKey(archive, key, article.msgid)
@@ -407,7 +407,7 @@
                 else:
                     parent = self.database.getArticle(self.archive,
                                                     article.parentID)
-                    article.threadKey = parent.threadKey+article.date+'-'
+                    article.threadKey = parent.threadKey + article.date + '/' + article.msgid + '-'
                 self.database.setThreadKey(self.archive,
                     (article.threadKey, article.msgid),
                     msgid)
@@ -615,9 +615,9 @@
             article.parentID = parentID = self.get_parent_info(arch, article)
             if parentID:
                 parent = self.database.getArticle(arch, parentID)
-                article.threadKey = parent.threadKey + article.date + '-'
+                article.threadKey = parent.threadKey + article.date + '/' + article.msgid + '-'
             else:
-                article.threadKey = article.date + '-'
+                article.threadKey = article.date + '/' + article.msgid + '-'
             key = article.threadKey, article.msgid
 
             self.database.setThreadKey(arch, key, article.msgid)
